### PR TITLE
Add button to delete firmware

### PR DESCRIPTION
**Pull request by ConnorRigby**
Only being able to delete firmware from the command-line was sort of annoying

### CODE REVIEW

**Comment by ConnorRigby** [comment]
@mobileoverlord I've added some error checking to the controller. I only used a simple `put_flash` to display a generic error.

File: apps/nerves_hub_www/lib/nerves_hub_www_web/controllers/firmware_controller.ex [hunk 1]
```diff
@@ -87,4 +87,13 @@ defmodule NervesHubWWWWeb.FirmwareController do
       {:error}
     end
   end
+
+  def delete(%{assigns: %{current_org: org, product: product}} = conn, %{"id" => id}) do
+    {:ok, firmware} = Firmwares.get_firmware(org, id)
+    :ok = Firmwares.delete_firmware(firmware)
```
Comment (mobileoverlord) [inline apps/nerves_hub_www/lib/nerves_hub_www_web/controllers/firmware_controller.ex]: The assumption of `:ok` will raise if things are not okay. When I run this locally, any firmware that has a constraint has a constraint error that is raised. We need to add the proper validation to the ecto changeset for the invalidating constraints so they are returned as ecto errors and we need this to capture those errors and put them in the flash when they happen.

**Comment by fhunleth** [comment]
@ConnorRigby Would you be able to get back to this? This seems useful and I'm hoping that the conflicts aren't too bad.

**Comment by ConnorRigby** [comment]
Yeah i think there were some changes that will make this code much easier to implement. Will take a look today.

**Comment by ConnorRigby** [comment]
I've rebased and updated this feature. I think it should be ready now;

CC: @fhunleth @mobileoverlord

**Pull request merged**